myst_parser/mocking.py

```diff
@@ -34,6 +34,16 @@ class MockInliner:
     This is parsed to role functions.
     """
 
+    # Override the ``__class__`` attribute to enable runtime type checking.
+    # See https://beartype.readthedocs.io/en/latest/faq/#mock-types.
+    @property
+    def __class__(self) -> type:
+        return Inliner
+
+    @__class__.setter
```

Comment:
This is not (yet) in the Beartype documentation, but the explanation for it is in https://github.com/beartype/beartype/pull/485.